framework/db/pgsql/EncoderTrait.php

```diff
@@ -0,0 +1,105 @@
+<?php
+namespace yii\db\pgsql;
+
+/**
+ * Trait contains methods to convert php arrays to PostgreSQL arrays and otherwise
+ *
+ * @author Ievgen Sentiabov <ievgen.sentiabov@gmail.com>
+ */
+trait EncoderTrait
+{
+    /**
+     * Convert php integer array to postgres array
+     * @param mixed $value
+     * @return string
+     */
+    public function arrayIntEncode($value)
+    {
+        $data = array_map('intval', (array)$value);
+        return '{'.implode(',', $data).'}';
+    }
+
+    /**
+     * Convert postgres integer array to php array
+     *
+     * @access public
+     * @param $value
+     * @return array
+     */
+    public function arrayIntDecode($value)
+    {
+        $value = str_replace(['{', '}'], '', $value);
+        if ($value === '') {
+            return [];
+        }
+        $data = explode(',', $value);
+        return array_map('intval', $data);
+    }
+
+    /**
+     * Convert php string array to postgres text array
+     *
+     * @access public
+     * @param mixed $value
+     * @return string
+     */
+    public function arrayTextEncode($value)
+    {
+        return '{'.implode(',', (array)$value).'}';
```

Comment:
Are you listen or not?
Current impelemtation does NOT allow explict set of `null` value to the column, which developer may require.
```php
$command->update('mytable', ['array_field' => null], 'id=1')->execute();
```
produces:
```sql
UPDATE ... SET (array_field, {})
```
while it should produce:
```sql
UPDATE ... SET (array_field, NULL)
```